spark2.4/src/main/scala/Inserter.scala

```diff
@@ -0,0 +1,83 @@
+import java.util.Properties
+import java.util.concurrent.{BlockingQueue, TimeUnit}
+import java.util.concurrent.atomic.AtomicBoolean
+
+import org.apache.kafka.clients.producer.KafkaProducer
+import org.apache.kafka.clients.producer.ProducerConfig
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.apache.kafka.common.serialization.IntegerSerializer
+import org.apache.kafka.common.serialization.LongSerializer
+import org.apache.spark.sql.types.StructType
+import com.splicemachine.spark2.splicemachine.SplicemachineContext
+import com.splicemachine.spark2.splicemachine.SplicemachineContext.RowForKafka
+
+class Inserter(
+    id: String,
+    spliceUrl: String, 
+    spliceKafkaServers: String, 
+    spliceKafkaPartitions: String,
+    spliceTable: String, 
+    dfSchema: StructType,
+    taskQueue: BlockingQueue[(Seq[RowForKafka], Long, String)],
+    batchCountQueue: BlockingQueue[Long],
+    processing: AtomicBoolean
+  )
+  extends Runnable {
+  
+  val nsds = new SplicemachineContext(Map(
+    "url" -> spliceUrl,
+    "KAFKA_SERVERS" -> spliceKafkaServers,
+    "KAFKA_TOPIC_PARTITIONS" -> spliceKafkaPartitions,
+    "USE_FLOW_MARKERS" -> "true"
+  ))
+  
+  nsds.setTable(spliceTable, dfSchema)
+
+  val props = new Properties
+  props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, spliceKafkaServers)
+  props.put(ProducerConfig.CLIENT_ID_CONFIG, "spark-producer-ssds-metrics-"+java.util.UUID.randomUUID() )
+  props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, classOf[IntegerSerializer].getName)
+  props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, classOf[LongSerializer].getName)
+
+  val metricsProducer = new KafkaProducer[Integer, Long](props)
+  val metricsTopic = "ssds-metrics"
+  
+  override def run(): Unit =
+    do {
+      try {
+        // From a LinkedBlockingQueue, get object containing topicname and rcd count
+        val task = taskQueue.poll(100L, TimeUnit.MILLISECONDS)
+        if (task != null && task._2 > 0) {
```

Comment:
Should there be any retry logic here in case of a timeout?  Could a temporary network issue cause us to hit the 100 ms timeout, in which case we could maybe print a warning message and retry a few times?